portfolio/src/main/webapp/script.js

```diff
@@ -39,9 +39,24 @@ function toggleDetailsButton() {
   }
 }
 
-function getHelloMessage() {
-    fetch('/data').then(response => response.text()).then((message) => {
-        document.getElementById('message-container').innerText = message;
+// gets the data from server sides
+function getServerData() {
+    fetch('/data')  
+    .then(response => response.json()) 
+    .then((data) => {
+        const dataListElement = document.getElementById('comments');
+        dataListElement.innerHTML = '';
```

Comment:
Suggesting extracting logic of populating "comments" and merge it with createListElement to put visualization logic together.

Reply:
alright. i made some changes.